### ISSUE TITLE
Can worker use more than one GPU?

### ISSUE THREAD

**Issue by kernelogic** [filecoin-project/lotus#2451]
I throw in two GTX 1060 but noticed in nvidia-smi only one is being used. Although `lotus-storage-miner workers list` shows two being used.

Worker 1, host dev-X570-UD
	CPU:  [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||] 16 core(s) in use
	RAM:  [|||||||||||||||||                                               ] 28% 35.5 GiB/125.8 GiB
	VMEM: [||||                                                            ] 8% 35.5 GiB/413 GiB
	GPU: GeForce GTX 1060 3GB, used
	GPU: GeForce GTX 1060 3GB, used
Thu Jul 16 20:17:01 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.100      Driver Version: 440.100      CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:04:00.0 Off |                  N/A |
| 55%   66C    P2    89W / 120W |   2983MiB /  3019MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 106...  Off  | 00000000:09:00.0 Off |                  N/A |
| 31%   40C    P8    10W / 120W |     10MiB /  3017MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      7001      C   lotus-seal-worker                           2973MiB |
+-----------------------------------------------------------------------------+

**Comment by kernelogic** [comment]
Never mind. In committing phase it is using all GPUs.

**Comment by harlovski** [comment]
Hi @kernelogic!
I have same GPUs and wanna try mine filecon .Maybe you can answer some of my questions?
1. Is it possible to mine with GTX 1060 3Gb today?
2. Are two GPUs enough? Or it will be serious bottle neck for r7 3700/128 ram worker?
I know that it will be better to use more powerfull gpu like 1080ti+, but may it be possible to start with these GPUs?

**Comment by kernelogic** [comment]
@harlovski You must use a GPU with more than 8GB VRAM. Meaning only the 1080TI, 2080TI, 3080, 3090 will work.

No matter how many 1060 3GB you put in, it wont work.

**Comment by harlovski** [comment]
I got it, thanks a lot! Maybe you can answer my another question?
How CPU performance, RAM volume, total ssd volume and GPU depend on each other? 
For example, I will start with consumer CPU, like ryzen r7 3700 / 128Gb RAM max / rtx 3060 / 1tb SSD. Then I will raise ssd volume to 4tb, 8tb etc.
Is there a chance that very soon 128GB ram will be not enough and I will have to change platform to threadripper? Or 8 cores will be the bottleneck? 
So basically I wanna know, is it ok to start with this setup, or it is meaningless and I need to choose hi-end platform with 256+ RAM and raise it with ssd volume.

**Comment by kernelogic** [comment]
You can seal 1 or 2 sectors concurrently with 128G RAM (Ryzen) and 3 to 4 sectors with 256G RAM (Threadripper). It's all depends on your desire for speed.

Remember you won't get any block reward before you can seal 10TB, and won't have much luck winning blocks between 10TB and 100TB.

**Comment by harlovski** [comment]
Thanks, the CPU and RAM size just determine mine speed, basicaly r7 and 128gb ram is ok. 
I don't understand what do you mean this:
> and won't have much luck winning blocks between 10TB and 100TB.
For example i will have 8TB HDD storage. With my rig (8 cores/128GB) i will fill this storage with speed 128GB per day, right? It wiil be 63 days before I will have any chance to get first reward. With 8TB HDD store my chances to get the block will be something like 1 block per month. Am I right? Or maybe somewhere there is a calculatore or something to help me count?

I have already asked you so much. Thank you! I hope it is alright. Maybe it's better to ask you via email?

**Comment by kernelogic** [comment]
The minimum starting point of winning blocks is 10TB. So you'll need 80 days to reach 10TB. After that your luck is proportional to your capacity. At 200TB you are probably able to win 1 block per day. I am at 20TB and I won a block every 20 days or so.
